AWS Lambda node documentation (#1489)
thingsboard/thingsboard.github.io: _includes/docs/user-guide/rule-engine-2-0/ce-external-nodes.md

* TOC
{:toc}

## AWS Lambda Node

<table  style="width:250px;">
   <thead>
     <tr>
	 <td style="text-align: center"><strong><em>Since TB Version 3.7.1</em></strong></td>
     </tr>
   </thead>
</table> 

![image](/images/user-guide/rule-engine-2-0/nodes/external-aws-lambda.png)

> Node publish messages to AWS Lambda (Amazon Lambda).

Reviewer: Node publishes messages to AWS Lambda, a service that lets you run code without provisioning or managing servers. It sends messages using a RequestResponse invocation type. The node uses a pre-configured client and specified function to run.

**Configuration**

// todo: add config image

Function configuration

> - **Function name** is required to specify which AWS Lambda function will be invoked and executed.

Reviewer: required parameter to specify which AWS Lambda function should be invoked.

> - **Qualifier** is used to specify a version or alias of the Lambda function to invoke. By default, Lambda invokes the **$LATEST** version, which is the latest version of the function.

Reviewer: optional parameter to specify a version or alias of the Lambda function. If the qualifier is not specified, the default qualifier "$LATEST" will be used.

> - **Invocation type** specifies how the Lambda function is invoked. You can choose from the following options: RequestResponse, Event and DryRun. More information on invocation types can be found [here](https://docs.aws.amazon.com/lambda/latest/api/API_Invoke.html#API_Invoke_RequestSyntax).

Reviewer: to remove: "Invocation type"

AWS Credentials

- **AWS Access Key ID** and **AWS Secret Access Key** are the credentials of an AWS IAM User with programmatic access.
  More information on AWS access keys can be found [here](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_access-keys.html).
- **AWS Region** must correspond to the one in which the Lambda function is created. Current list of AWS Regions can be found [here](https://docs.aws.amazon.com/general/latest/gr/rande.html).

Advanced settings

> - **Connection timeout**: amount of time to wait (in milliseconds) when initially establishing a connection before giving up and timing out.

Reviewer: ```suggestion
- **Connection timeout**: amount of time to wait (in seconds) when initially establishing a connection before giving up and timing out.
```

  A value of 0 means infinity, and is not recommended.

> - **Request timeout**: amount of time to wait (in milliseconds) for the request to complete before giving up and timing out.

Reviewer: ```suggestion
- **Request timeout**: amount of time to wait (in seconds) for the request to complete before giving up and timing out.
```

> - **Tell Failure if AWS Lambda function execution raises exception**: if enabled, forces failure of message processing if AWS Lambda function execution raises exception. If disabled, the error information is added to the response payload, and the message continues via the success route.

Reviewer: ```suggestion
- **Tell Failure if AWS Lambda function execution raises exception**: if enabled, forces failure of message processing if AWS Lambda function execution raises exception. If disabled, the error information is added to the response payload, and the message will be routed via success chain.
```

> > **Note**: **Function name** and **Qualifier** fields supports templatization.

Reviewer: ```suggestion
> **Note**: **Function name** and **Qualifier** fields support templatization.
```

**Output**
- **Success**: If message was processed successfully.

> - **Failure**: If error happened during message processing or Lambda function execution raises exception, if **Tell Failure if AWS Lambda function execution raises exception** is enabled.

Reviewer: If an error occurs during message processing or if the AWS Lambda function execution raises an exception, and the option 'Tell Failure if AWS Lambda function execution raises exception' is enabled.
